cmd/scaffold.go

```diff
@@ -722,6 +743,10 @@ func (fnb *FlowNodeBuilder) Initialize() NodeBuilder {
 		fnb.RegisterBadgerMetrics()
 	}
 
+	if fnb.adminAddr != NotSet {
```

Comment:
In this case we would just ignore the `adminHttpAddr`. I can add validation for this although I don't think it's really that necessary?

The `adminHttpAddr` only makes sense if `adminAddr` is set. We cannot have an http server if the underlying gRPC service doesn not even exist.

Reply:
First, I think you can serve GRPC over a Unix domain socket rather than a TCP socket (and in this context, this might actually not even be such a bad idea).

But more to my original point, you can interpret the presence of an `adminHttpAddr` alone as an authorization to bind to a random port for the GRPC endpoint, especially if you document that it is the contract.

Reply:
Sure, I guess we can do that.

Reply:
Actually on second thought, we cannot just choose a random port because we also need to know which interface to bind to.

As I mentioned in the previous comment, I'm not really a fan of limiting the interface to localhost only.

Reply:
The thing I'm advocating for here is safe defaults.
- if I have not mentioned an IP for either server, I bind to localhost,
- if I have not mentioned a port for either server, it can be random

Reply:
I think we can have a safe default, and that an address parameter that specifies, e.g. just a port, can be interpreted as a default binding to localhost.
With that said, we distribute this through Docker, which won't expose an unexported port, so at this stage it's a nit.